src/ebonite/ext/prebuild.py

```diff
@@ -0,0 +1,62 @@
+import os
+import re
+import shutil
+import sys
+import tempfile
+
+from jinja2 import Environment, FileSystemLoader
+
+from ebonite.build.docker import (RemoteDockerRegistry, create_docker_client, image_exists_at_dockerhub,
+                                  login_to_registry, repository_tags_at_dockerhub)
+from ebonite.utils.log import logger
+
+
+def prebuild_image(prebuild_path, name_template, python_version, *, push=False):
+    tag = name_template.format(python_version)
+    if image_exists_at_dockerhub(tag):
+        logger.info('Skipped building image %s: already exists', tag)
+        return
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        context_dir = os.path.join(tmpdir, 'context')
+        logger.info('Building image %s on %s ...', tag, context_dir)
+
+        shutil.copytree(prebuild_path, context_dir)
+        _generate_dockerfile(context_dir, python_version)
+
+        try:
+            with create_docker_client() as client:
```

Comment:
this looks a lot like docker_builder code. cant we reuse it by implementing ServerProvider? We can then reuse it for different servers

Reply:
1. Obviously we can reuse builder code but the main point was to detach this script from Ebonite completely in future.
1. I expect this script to be reused for all servers. That's why I put it there and not in `flask` package.

Reply:
but it still will depend on ebonite, so it can use it internals. but ok